e2e2/test/cases/nvidia/main_test.go

```diff
@@ -73,12 +86,59 @@ func TestMain(m *testing.M) {
 			}
 			return ctx, nil
 		},
+		func(ctx context.Context, config *envconf.Config) (context.Context, error) {
+			clientset, err := kubernetes.NewForConfig(cfg.Client().RESTConfig())
+			if err != nil {
+				return ctx, err
+			}
+			nodes, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return ctx, err
+			}
+			if *efaEnabled {
+				err := fwext.ApplyManifests(cfg.Client().RESTConfig(), efaDevicePluginManifest)
+				if err != nil {
+					return ctx, err
+				}
+				ds := appsv1.DaemonSet{
+					ObjectMeta: metav1.ObjectMeta{Name: "efa-device-plugin-daemonset", Namespace: "kube-system"},
+				}
+				err = wait.For(fwext.NewConditionExtension(cfg.Client().Resources()).DaemonSetReady(&ds),
+					wait.WithTimeout(time.Minute*5))
+				if err != nil {
+					return ctx, err
+				}
+			}
+			if *nodeType != "" {
+				for _, v := range nodes.Items {
+					if v.Labels["node.kubernetes.io/instance-type"] == *nodeType {
+						nodeCount++
+						gpu := v.Status.Capacity["nvidia.com/gpu"]
+						gpuCount = int(gpu.Value())
```

Comment:
do we care about detecting whether this is the correct value? how could we tell what's going on if the nvidia device plugin is malfunctioning here?

Reply:
In my opinion, we should do this because it allows us to test the NCCL operation more accurately. We need to set up the `-np` flag and `slotsPerWorker`. If the NVIDIA device plugin malfunctions, the test will fail at [here](https://github.com/aws/aws-k8s-tester/blob/main/e2e2/test/cases/nvidia/main_test.go#L69-L73).

Reply:
ok, as long as we can view the details/parameters for the run its fine